tensorboard/tools/whitespace_hygiene_test.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Check for superfluous whitespace at ends of lines.
+
+Keeps diffs clean and persnickety developers happy.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import os
+import subprocess
+import sys
+
+
+# Remove files from this list as whitespace errors are fixed.
```

Comment:
Are any of these ones that should actually be longstanding exceptions?  If not, why not just commit the whitespace fixes to these now and avoid ever checking in the exceptions list support and stale exceptions detection?

Reply:
> Are any of these ones that should actually be longstanding exceptions?

Nope. (The only potential genuine exception that I could think of would
be Markdown hard-breaks, which should rarely be used anyway.)

> why not just commit the whitespace fixes to these now and avoid ever
> checking in the exceptions list support

Two benefits:

  - we avoid a megacommit that kills all the blame (existing such
    commits, like e1adbef8 and friends, are perennially frustrating in
    everyday TensorBoard development); and
  - in the case that we _do_ need a genuine exception (say, some strange
    data format for a plugin’s test data), the path of least resistance
    will be to mark it as an exception rather than just disabling the
    test because there’s an urgent deadline.

Reply:
Running your script shows that there are only 88 affected individual lines.  I don't think an 88-line delta is a megacommit and it's certainly nothing like a 11,000 line one.  Blame impact would be localized only to affected lines.

(Though, even if we were talking about reformatting en masse (e.g. buildifiering all our build files) I think it's better to just get it over with.  Doing it one touched line at a time will never actually fix everything so you never achieve the uniformity that's the desired goal anyway.  Doing it file-by-file means that you still clobber the blame but now there's hundreds of such commits instead of one.)

The second benefit is entirely speculative.

Reply:
Good point: I was scared off by the file count and didn’t look at the
delta. That does make it much less bad. I’m convinced; will send out a
fix.

> The second benefit is entirely speculative.

Certainly.